tests/components/recorder/test_init.py

```diff
@@ -24,99 +21,69 @@
 from .common import wait_recording_done
 
 from tests.async_mock import patch
-from tests.common import (
-    async_fire_time_changed,
-    get_test_home_assistant,
-    init_recorder_component,
-)
-
-
-class TestRecorder(unittest.TestCase):
-    """Test the recorder module."""
-
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        init_recorder_component(self.hass)
-        self.hass.start()
-        self.addCleanup(self.tear_down_cleanup)
+from tests.common import async_fire_time_changed, get_test_home_assistant
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
 
-    def test_saving_state(self):
-        """Test saving and restoring a state."""
-        entity_id = "test.recorder"
-        state = "restoring_from_db"
-        attributes = {"test_attr": 5, "test_attr_10": "nice"}
-
-        self.hass.states.set(entity_id, state, attributes)
+def test_saving_state(hass, hass_recorder):
+    """Test saving and restoring a state."""
+    hass = hass_recorder()
 
-        wait_recording_done(self.hass)
+    entity_id = "test.recorder"
+    state = "restoring_from_db"
+    attributes = {"test_attr": 5, "test_attr_10": "nice"}
 
-        with session_scope(hass=self.hass) as session:
-            db_states = list(session.query(States))
-            assert len(db_states) == 1
-            assert db_states[0].event_id > 0
-            state = db_states[0].to_native()
+    hass.states.set(entity_id, state, attributes)
 
-        assert state == _state_empty_context(self.hass, entity_id)
+    wait_recording_done(hass)
 
-    def test_saving_event(self):
-        """Test saving and restoring an event."""
-        event_type = "EVENT_TEST"
-        event_data = {"test_attr": 5, "test_attr_10": "nice"}
+    with session_scope(hass=hass) as session:
+        db_states = list(session.query(States))
+        assert len(db_states) == 1
+        assert db_states[0].event_id > 0
+        state = db_states[0].to_native()
 
-        events = []
+    assert state == _state_empty_context(hass, entity_id)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6294088)